### DISCUSSION TITLE
Using Multi-Site capabilities in Craft CMS v3.9 where `HTTP_HOST` isn't accurate

### DISCUSSION THREAD

**Discussion by GaryReckard** [craftcms/cms#14544] [Q&A]
I've got a site that is behind a Silverline WAF and Azure Front Door, it runs in a docker container, and for reasons I don't 100% understand, the actual hostname isn't getting passed through to Craft. If I output `$_SERVER['HTTP_HOST']` it is just showing the IP address. (Same for `$_SERVER['SERVER_NAME']`)

I believe we've got things working for the single site by setting `$_SERVER['HTTP_HOST']` based on an `.env` variable in the production environment. But now we're adding three new sites. The new sites are using unique subdirectories, such as /en-fr/ and /fr-fr/, and this is working great locally and on staging. But on production, visiting those subdirectories throws a 404.

I've tried a few things after some googling, but none of it has resolved the issue. I've tried setting the `siteUrl` value for each site manually in the `general.php` config:
```
        'siteUrl' => [
            'enUs' => $siteUrl,
            'enGb' => $siteUrl.'/en-gb/',
            'enFr' => $siteUrl.'/en-fr/',
            'frFr' => $siteUrl.'/fr-fr/',
        ],
```

Also, I originally had the new sites configured with a baseUrl like `baseUrl: '@web/fr-fr/'`. (The primary site uses `baseUrl: 'https://$SITE_URL'`). That wasn't working, so I tried changing that to something like `baseUrl: '@web_frFr/'`
and setting up these aliases in `general.php`:
```
        'aliases' => [
            '@web' => $siteUrl,
            '@web_enUs' => $siteUrl,
            '@web_enGb' => $siteUrl.'/en-gb/',
            '@web_enFr' => $siteUrl.'/en-fr/',
            '@web_frFr' => $siteUrl.'/fr-fr/',
        ],
```
but no luck there.

These attempts probably expose my lack of understanding + confusion about how all this really works...

Wondering if anyone has any suggestions on how to get this multi-site stuff to work in my type of setup. Sidenote: we do have plans to upgrade to Craft v4, but that's still maybe months away. We need to get these new regional sites up ASAP.

Thanks!

**Comment by angrybrad** [comment]
I can't think of (or find) anywhere in Yii or Craft where we write to either HTTP_HOST or SERVER_NAME.

We _read_ from those values, sure. https://github.com/yiisoft/yii2/blob/master/framework/web/Request.php#L754-L777

I’ve never used Silverline WAF or Azure Front Door, but as a starting point, I’d focus on those to see if they are injecting values into those headers that Craft/Yii is just reading from.
